app/models/recipe.rb

```diff
@@ -1,8 +1,15 @@
 class Recipe < ApplicationRecord
   validates :name, presence: true
-  validate :must_link_be_correct 
+  validate :must_link_be_correct,:must_level_be_correct
+  
+  def must_level_be_correct
+    return unless level
+    errors.add(:level,"the number must be between 1 and 5") unless (1..5).include? level
+  end  
 
   def must_link_be_correct 
+    return unless video_link
     errors.add(:video_link,"This link is uncorrect") unless video_link =~ URI::regexp
```

Comment:
```suggestion
    return unless video_link
    
    errors.add(:video_link,"This link is uncorrect") unless video_link =~ URI::regexp
```

besides you can use rails validation with the format.